framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -477,6 +485,38 @@ public void updateInferredMethodReturnType(
         AnnotatedTypeMirror rhsATM = atf.getAnnotatedType(retNode.getTree().getExpression());
         helper.updateAnnotationSetInScene(
                 method.returnType, atf, jaifPath, rhsATM, lhsATM, TypeUseLocation.RETURN);
+
+        // Now, update overridden methods based on the implementation we just saw.
+        // If the overridden method is abstract, then WPI cannot update it directly,
```

Comment:
Clarify "directly".  I think you mean that there are no calls that resolve to it.